packages/SwingSet/src/controller/initializeSwingset.js

```diff
@@ -341,17 +342,15 @@ export async function initializeSwingset(
   // worker (e.g. which exercise XS heap snapshots, or metering) can
   // override the env var, so they won't break under `yarn test:xs`.
 
-  let defaultManagerType =
-    config.defaultManagerType || SWINGSET_WORKER_TYPE || 'local';
-  if (defaultManagerType === 'xs-worker') {
+  if (!config.defaultManagerType) {
+    config.defaultManagerType = SWINGSET_WORKER_TYPE || 'local';
+  }
+  if (config.defaultManagerType === 'xs-worker') {
     // 'xs-worker' is an alias accepted for now
-    defaultManagerType = 'xsnap'; // but 'xsnap' is preferred
+    config.defaultManagerType = 'xsnap'; // but 'xsnap' is preferred
   }
-  assert(
-    defaultManagerType === 'local' || defaultManagerType === 'xsnap',
-    `unknown manager type ${defaultManagerType}`,
-  );
-  config.defaultManagerType = defaultManagerType;
+  const ManagerTypeShape = M.or('local', 'xsnap');
```

Comment:
I was going to say this should go in typeGuards. Turns out we had one already =)

https://github.com/Agoric/agoric-sdk/blob/b11559310bdd701d0d0e2a856ccf7072ceffed99/packages/SwingSet/src/typeGuards.js#L4

And bonus, we have a `SwingSetConfig` shape in there too. If you're feeling thorough, you could use that to validate all these mutations. Also rename to `SwingSetConfigShape` to match emerging conventions. (Not _requesting_ because I'm not sure the shape is accurate for all swingset configs.)

Reply:
aaand.. now it's time for me to fight with `tsc` again.

```js
  if (!config.defaultManagerType) {
    config.defaultManagerType = SWINGSET_WORKER_TYPE || 'local';
  }
```

means that I'm storing `SWINGSET_WORKER_TYPE` (which is an arbitrary string?) into `config.defaultManagerType` (which has a type), so tsc complains:

```
src/controller/initializeSwingset.js:347:5 - error TS2322: Type 'string' is not assignable to type 'ManagerType | undefined'.

347     config.defaultManagerType = SWINGSET_WORKER_TYPE || 'local';
```

I imagine that if I made it past the `mustMatch` check, I'd have something which will be accepted by `config.defaultManagerType`. I'll try doing it in a temp variable again.

Reply:
nope, `mustMatch` doesn't provide a type constraint?

Reply:
This works:
```js
  let defaultManagerType =
    config.defaultManagerType || SWINGSET_WORKER_TYPE || 'local';
  if (defaultManagerType === 'xs-worker') {
    // 'xs-worker' is an alias accepted for now
    defaultManagerType = 'xsnap'; // but 'xsnap' is preferred
  }
  switch (defaultManagerType) {
    case 'local':
    case 'xsnap':
      config.defaultManagerType = defaultManagerType;
      break;
    default:
      Fail`unknown manager type ${defaultManagerType}`;
  }
  mustMatch(config.defaultManagerType, ManagerType);
```